tests/integration/workspace_access/test_redash.py

```diff
@@ -0,0 +1,115 @@
+    # More details here: https://databricks.atlassian.net/browse/ES-992619
+    @retried(on=[AssertionError], timeout=timedelta(minutes=10))
+    def check_permissions_for_backup_group():
+        logger.info("check_permissions_for_backup_group()")
+
+        query_permissions = redash_permissions.load_as_dict(sql.ObjectTypePlural.QUERIES, query.id)
+        assert group_info.temporary_name in query_permissions
+        assert user.display_name in query_permissions
+        assert sql.PermissionLevel.CAN_RUN == query_permissions[group_info.temporary_name]
+        assert sql.PermissionLevel.CAN_RUN == query_permissions[user.display_name]
+
+    check_permissions_for_backup_group()
+
+    group_manager.reflect_account_groups_on_workspace()
+
+    @retried(on=[AssertionError], timeout=timedelta(minutes=1))
+    def check_permissions_after_replace():
+        logger.info("check_permissions_after_replace()")
+
+        query_permissions = redash_permissions.load_as_dict(sql.ObjectTypePlural.QUERIES, query.id)
+        assert group_info.name_in_account not in query_permissions
+        assert group_info.temporary_name in query_permissions
+        assert user.display_name in query_permissions
+        assert sql.PermissionLevel.CAN_RUN == query_permissions[group_info.temporary_name]
+        assert sql.PermissionLevel.CAN_RUN == query_permissions[user.display_name]
+
+    check_permissions_after_replace()
+
+    permission_manager.apply_group_permissions(state)
+
+    @retried(on=[AssertionError], timeout=timedelta(minutes=1))
+    def check_permissions_for_account_group():
+        logger.info("check_permissions_for_account_group()")
+
+        query_permissions = redash_permissions.load_as_dict(sql.ObjectTypePlural.QUERIES, query.id)
+        assert group_info.name_in_account in query_permissions
+        assert user.display_name in query_permissions
+        assert sql.PermissionLevel.CAN_RUN == query_permissions[group_info.name_in_account]
+        assert sql.PermissionLevel.CAN_RUN == query_permissions[group_info.temporary_name]
+        assert sql.PermissionLevel.CAN_RUN == query_permissions[user.display_name]
+
+    check_permissions_for_account_group()
+
+    group_manager.delete_original_workspace_groups()
+
+    @retried(on=[AssertionError], timeout=timedelta(minutes=1))
+    def check_permissions_after_backup_delete():
+        logger.info("check_table_permissions_after_backup_delete()")
+
+        query_permissions = redash_permissions.load_as_dict(sql.ObjectTypePlural.QUERIES, query.id)
+        assert group_info.temporary_name not in query_permissions
+        assert group_info.name_in_account in query_permissions
+        assert user.display_name in query_permissions
+        assert sql.PermissionLevel.CAN_RUN == query_permissions[group_info.name_in_account]
+        assert sql.PermissionLevel.CAN_RUN == query_permissions[user.display_name]
+
+    check_permissions_after_backup_delete()
```

Comment:
this test is too big and going to be flaky for everyone, staggering the merges. please simplify/rewrite it. Look at `test_permissions_for_secrets` as the example short/focused/quick test.

Reply:
simplified a bit but we need to go thru the flow of crawling, then renaming and then applying permissions. Otherwise we cannot catch the issue we are trying to solve here.

Reply:
updated